owncloudApp/src/main/java/com/owncloud/android/ui/adapter/SharePublicLinkListAdapter.kt

```diff
@@ -0,0 +1,86 @@
+package com.owncloud.android.ui.adapter
+
+/**
+ * ownCloud Android client application
+ *
+ * @author David González Verdugo
+ * @author Christian Schabesberger
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
+ */
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import com.owncloud.android.R
+import com.owncloud.android.shares.db.OCShare
+import com.owncloud.android.utils.PreferenceUtils
+import kotlinx.android.synthetic.main.share_public_link_item.view.*
+import java.util.ArrayList
+
+/**
+ * Adapter to show a list of public links
+ */
+class SharePublicLinkListAdapter(
+    private val mContext: Context, resource: Int, private val mPublicLinks: ArrayList<OCShare>?,
+    private val mListener: SharePublicLinkAdapterListener
+) : ArrayAdapter<OCShare>(mContext, resource) {
+
+    override fun getCount(): Int {
+        return mPublicLinks!!.size
```

Comment:
it can be null!!!

Reply:
override fun getCount(): Int = mPublicLinks?.size ?: 0

better?

Reply:
Yes, done